### PR TITLE
Fix issue with gathering with NVCategory and nulls

### PR DESCRIPTION
**Pull request by shwina**

### CODE REVIEW

File: cpp/tests/replace/replace-nulls_tests.cu [hunk 1]
```diff
@@ -54,6 +55,20 @@ void ReplaceNullsScalar(cudf::test::column_wrapper<T> input,
   EXPECT_TRUE(expected == result);
 }
 
+template <typename T>
+void ReplaceNullsScalarInplace(cudf::test::column_wrapper<T> input,
+                               cudf::test::scalar_wrapper<T> replacement_value,
+                               cudf::test::column_wrapper<T> expected)
+{
+  gdf_column result;
+  print_gdf_column(input);
```
Comment (shwina) [inline cpp/tests/replace/replace-nulls_tests.cu]: Need to get rid of these

File: CHANGELOG.md [hunk 2]
```diff
@@ -148,10 +148,13 @@
 - PR #1965 Parquet Reader: Fix duplicate index column when it's already in `use_cols`
 - PR #2033 Add pip to conda environment files to fix warning
 - PR #2028 CSV Reader: Fix reading of uncompressed files without a recognized file extension
+<<<<<<< HEAD
```
Comment (kkraus14) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -148,10 +148,13 @@
...
+- PR #2073 Fix an issue when gathering columns with NVCategory and nulls
 - PR #2053 cudf::apply_boolean_mask return empty column for empty boolean mask
 - PR #2069 Fix JNI code to use read_csv and read_parquet APIs
 - PR #2071 Fix bug with unfound transitive dependencies for GTests in Ubuntu 18.04
 
+
```
Comment (kkraus14) [inline CHANGELOG.md]:
```suggestion
```

**Comment by karthikeyann** [comment]
will this PR fix issue https://github.com/rapidsai/cudf/issues/2000 ?

**Pull request merged**